src/types.js

```diff
@@ -11,7 +11,7 @@ import type {Context as KoaContext} from 'koa';
 export type Token<T> = {
   (): T,
   optional: () => void | T,
-  stack: string,
+  stacks: Array<{type: string, stack: string}>,
```

Comment:
Can we have `type` be an enum here rather than just a string?

Reply:
Oh, good idea!